src/debug/telemetry/telemetry.c

```diff
@@ -94,6 +103,71 @@ static int perf_bitmap_is_bit_clear(struct perf_bitmap * const bitmap, size_t bi
 	return !val;
 }
 
+struct perf_data_item_comp *perf_data_getnext(void)
+{
+	int idx;
+	int ret = perf_bitmap_alloc(&performance_data_bitmap, &idx);
+
+	if (ret < 0)
+		return NULL;
+	/* original fw did not set the bits, but here we do it to not have to use
+	 * isFree() check that the bitarray does not provide yet. Instead we will use isClear
+	 * ,and always set bit on bitmap alloc.
+	 */
+	ret = perf_bitmap_setbit(&performance_data_bitmap, idx);
+	if (ret < 0)
+		return NULL;
+	return &perf_data_[idx];
+}
+
+int perf_data_free(struct perf_data_item_comp * const item)
+{
+	/* find index of item */
+	int idx = (item - perf_data_) / sizeof(*item);
+	int ret = perf_bitmap_clearbit(&performance_data_bitmap, idx);
+
+	if (ret < 0)
+		return -EINVAL;
+	ret = perf_bitmap_free(&performance_data_bitmap, idx);
+	if (ret < 0)
+		return -EINVAL;
```

Comment:
if this is supposed to return POSIX errors, just `return ret;` and `return 0` below